auth-server/src/test/java/com/makeurpicks/service/OAuthServiceTest.java

```diff
@@ -0,0 +1,199 @@
+package com.makeurpicks.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.contains;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.autoconfigure.orm.jpa.AutoConfigureTestDatabase;
+import org.springframework.security.oauth2.provider.ClientDetails;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.makeurpicks.AuthServerApplication;
+import com.makeurpicks.dao.ClientDetailsDAO;
+import com.makeurpicks.domain.OAuthClientDetails;
+import com.makeurpicks.domain.OAuthClientDetailsBuilder;
+import com.makeurpicks.exception.OAuthclientValidationException;
+import com.makeurpicks.exception.OAuthclientValidationException.OAuthClientExceptions;
+
+@RunWith(SpringJUnit4ClassRunner.class)
```

Comment:
[Hari 02/17/2017]: HI Powell, Generally I run as Junit test at project level rather than as class level. Thats how I tested it yesterday. I also tested by running Individual test case. That was also running. I can show a demo if needed. The reason why I run at project level is because
1. In actual deployments when we give maven goal for jUnit test, it executes all test cases.
2. Sometimes the autowiring did not work and the dependent class instances were not set. Due to this the test were failing ( Related to the next comment). That is why I run at project level.


[Hari 2/17/2017]: Hi Powell, I did run yesterday.  All test cases ran. 4/6 were mentioned as success. I am looking at why the other two were failing when they were coded in same manner as the one which were succeeded.